src/Command/QueueCommand.php

```diff
@@ -100,20 +80,27 @@ private function loop(InputInterface $input)
         $this->logger->debug($this->getName().' Loop start, listening to queue', ['queue' => $this->queue->getName()]);
         $item = $this->queue->dequeue();
         if ($item) {
-            $this->monitoring->startTransaction();
-            if ($entity = $this->transform($item)) {
-                try {
+            try {
+                $this->monitoring->startTransaction();
+                if ($entity = $this->transform($item)) {
                     $this->process($input, $item, $entity);
-                } catch (Throwable $e) {
-                    $this->logger->error("{$this->getName()}: There was an unknown problem processing {$item->getType()} ({$item->getId()})", [
-                        'exception' => $e,
-                        'item' => $item,
-                    ]);
-                    $this->monitoring->recordException($e, "Error in processing {$item->getType()} {$item->getId()}");
                 }
+                $this->monitoring->endTransaction();
+            } catch (BadResponse $e) {
+                // We got a 404 or server error.
+                $this->logger->error("{$this->getName()}: Item does not exist in API: {$item->getType()} ({$item->getId()})", [
+                    'exception' => $e,
+                    'item' => $item,
+                ]);
+            } catch (Throwable $e) {
+                $this->logger->error("{$this->getName()}: There was an unknown problem processing {$item->getType()} ({$item->getId()})", [
+                    'exception' => $e,
+                    'item' => $item,
+                ]);
+                $this->monitoring->recordException($e, "Error in processing {$item->getType()} {$item->getId()}");
+            } finally {
                 $this->queue->commit($item);
```

Comment:
Pre-existing, but we never retry?

Reply:
Discussing with @nlisgo we could introduce a configurable whitelist of exceptions that are allowed to retry. Requires looking at actual traffic and see which errors are frequently raised to put them into the list.